### ISSUE TITLE
Multiple brand Icons not loading

### ISSUE THREAD

**Issue by HarvsG** [home-assistant/brands#3753]
### The problem

![image](https://user-images.githubusercontent.com/11440490/194702287-d50a62b7-8cdb-4c11-9e0d-a2b3a8ab2632.png)

![image](https://user-images.githubusercontent.com/11440490/194702309-d881d492-2838-4719-bbf4-71c928c10719.png)

But the images are located at: `https://brands.home-assistant.io/brands/_/raspberry_pi/icon.png` not `https://brands.home-assistant.io/brands/_/raspberry/icon.png`

https://github.com/home-assistant/core/blob/e7b550685e451f830ebb1069c720c73381e6befb/homeassistant/generated/integrations.json#L3489

### What version of Home Assistant Core has the issue?

2022.10.1

### What type of installation are you running?

Home Assistant OS

### Integration causing the issue

raspberrypi

### Link to integration documentation on our website

https://www.home-assistant.io/installation/raspberrypi/

### Additional information
Also an issue with multiple others:
SensorPush
![image](https://user-images.githubusercontent.com/11440490/194702793-1e84afbf-75d6-49f0-9110-163f8f1f24c8.png)

![image](https://user-images.githubusercontent.com/11440490/194702815-a5e16b71-cb4b-4127-9e00-fc93f9a268a0.png)

Smart Meter Texas

![image](https://user-images.githubusercontent.com/11440490/194702867-a02593ca-aa68-43be-8e2b-a83c20e36a17.png)

![image](https://user-images.githubusercontent.com/11440490/194702876-169ce872-1a3e-466c-8f8a-58003877076d.png)

**Comment by frenck** [comment]
Moving this issue into the issue tracker right repository

**Comment by HarvsG** [comment]
OK. I put on core because I think the fix is likely renaming
https://github.com/home-assistant/core/blob/dev/homeassistant/brands/raspberry.json
to
https://github.com/home-assistant/core/blob/dev/homeassistant/brands/raspberry_pi.json

**Comment by frenck** [comment]
its brand image related, so the issue should be here, not core.

**Comment by HarvsG** [comment]
I wonder if it would be possible to write a test that detects issues like this, might be helpful with a lot of upcoming brand re-organisation.

**Comment by KevinCathcart** [comment]
devolo: recently added to brands folder in this repo, should be fixed soon.

locative: Completely missing from this repo.

smart_meter_texas: this has a logo, but no icon. They have no square logo variation, so the only option to create an icon is to extend the logo to be a square, and shrink to icon size. This is what other brands like Panasonic have done.

sensorpush: This has a logo and no icon. The logo is square looks like it is the correct size for an icon, so this looks liike the files were accidentally named wrong.

For Raspberry PI specifically, the cause is arguably the brand file violating the rules. Per https://developers.home-assistant.io/docs/creating_integration_brand/#domain the value in domain must match the file name. Currently the file is named raspberry.json and specifies raspberry_pi as the domain. Hassfess ignores the contents of the domain field, and uses the filename for generating the integrations list. (It should probably at least verify the domain matches file name rule.)

The image name as present in this repository's brands folder is raspberry_pi (matching the domain field, but not the brand json filename).

Fixing it over here would require symlinking core_brands/raspberry to core_brands/raspberry_pi (or moving core_brands/raspberry_pi, and updating a bunch of symlinks).

**Comment by frenck** [comment]
@KevinCathcart This issue was about the raspberry pi branding, not sure what all others have to do with that.

**Comment by KevinCathcart** [comment]
@frenck

The issue title is "Multiple brand Icons not loading", and issue body also lists sensorpush, and smart_meter_texas as additional examples. I also noticed the other two, and mentioned the status of them. All obviously should get addressed eventually, even though they could reasonably be broken out to separate issues.

**Comment by frenck** [comment]
Some are not broken... missing/not provided icons is not a bug. And yes, it should have been broken out probably, but the report is made against the raspberry pi, as per form that has been filled out.

**Comment by KevinCathcart** [comment]
Missing icons are not a bug, but they are an "issue" (in the sense of something that ideally ought to get addressed at some point). It is never intentional nor desirable for users to come across missing or broken images for built-in brands or integrations. It is something that does happen from time to time, but something the project does strive to avoid, hence things like the pull request templates suggesting that all new integrations create a pull request over here in brands to provide an icon and logo, or at least a link if a suitable icon for the brand is already available.

One should not blindly take the integration name and link fields that somebody types in the bug report as accurately limiting the scope of a bug report. People get those fields wrong all the time. It is always better to read the title and details of a bug report to accurately determine the scope. Here the full text of the issue provided seems to indicate to me that the poster is concerned about all of these, and has dug into one (Raspberry Pi) as a hopefully representative example, (although it appears that in actuality it is not so representative).

**Comment by frenck** [comment]
Sorry, maybe I didn't got across in my previous message, let me try again:

Missing icons are not bugs. Feel free to contribute them if you like, but it is not broken.

The image shown is a fallback by design. Again: a missing icon is not a bug.

Thanks.

../Frenck

**Comment by HarvsG** [comment]
Yes I originally opened the PR against raspberry pi in main because I spotted the `rapsberry` Vs `raspberry_pi` naming issue, which IMO is due to an error in folder name in core.

Then Frenk moved the issue here so I widened the scope to include all the missing brands I could find. Sorry if that has caused issues.
